### ISSUE TITLE
Command line?

### ISSUE THREAD

**Issue by vmiceli** [BudWalkerJava/DatCon#2]
Hello and thanks for the great software that DatCon is. If I wanted to automate file conversions with a script, can I use DatCon from command line?

Thanks!

Vincenzo

**Comment by Taylormade11** [comment]
+1 @BudWalkerJava Is this doable? I've been trying to figure out how to call the DatConvert functionality on a DatFile from command line or via bash script with no success

**Comment by BudWalkerJava** [comment]
Not currently. But, you could easily modify the code to create a command line app.

**Comment by Taylormade11** [comment]
@BudWalkerJava I have a lot of work to do before this is an easy task for me lol but with work, I'll get there. Do you know where I might start in learning to tackle this type of problem? I'm having trouble understanding how you are getting the filename from 'file being used' and then actually executing the conversion. Ideally, I'd like to be able to iterate through a directory of .dat files by passing the directory as an argument from the command line.

Thank you all so much for your hard work in providing everyone with this great software. It's been a huge help and has saved so much time!

**Comment by yvonperr** [comment]
The currently committed github version (3.5.0) has a lot of java compile errors (many because of lines commented out in AxesAndSigs.java, for example).  Is it possible to have a code version that compiles, or actually, the code version that actually produced the current DatCon?  It's hard to make assumptions and just uncomment lines, and so on.

**Comment by yvonperr** [comment]
How can we message Rowland Johnson?  Is it possible?

**Comment by BudWalkerJava** [comment]
> The currently committed github version (3.5.0) has a lot of java compile errors (many because of lines commented out in AxesAndSigs.java, for example). Is it possible to have a code version that compiles, or actually, the code version that actually produced the current DatCon? It's hard to make assumptions and just uncomment lines, and so on.

Try it now, I just updated the source.

**Comment by jothanna** [comment]
I'd also like to convert a larger amount of data from the command line. Has anyone changed the program in this respect?

**Comment by yvonperr** [comment]
@jothanna,

I did some work with DatCon for my then company a few months ago.  It works well for us, so I need to contribute my code back to the project.

If you can send me your preferred email address, I will send you an email explaining my modified jar file and source code, which will be attached

— Yvon Perreault

**Comment by jothanna** [comment]
@yvonperr, thank you for your help. Here my email address: j.redelstein(at)gmx.de.

**Comment by yvonperr** [comment]
Unfortunately, icloud won’t let me include any jar file or zip file.  Researching…

-Y-

**Comment by chkop** [comment]
I agree... A command line option would be great, specially to be able to batch process files.
Also an option for batch processing inside the gui could be helpful.

**Comment by munsanje** [comment]
Any progress on this front?

**Comment by munsanje** [comment]
@yvonperr are you still able to share your code?

**Comment by munsanje** [comment]
@yvonperr are you still able to share your code?

**Comment by yvonperr** [comment]
Yes I can.  Give me a few minutes...

**Comment by yvonperr** [comment]
Munsanje,

Can you provide your email address at yvon.at.mac.com?  I will share like I did with #jothanna.

**Comment by TopHatTommy2** [comment]
I'm hoping that this can be done with Python. I'm commenting more to follow the discussion but:
I am using a DJI Inspire 1 for Search and Rescue (SAR) operations. I need to be able to batch the files, then process the csv's to strip multiple columns of data, then I can process it into a table that can be turned into a shapefile for use in our GIS. I know Python. I don't know Java.

**Comment by nbeyne1998** [comment]
@yvonperr Is there a possibility to get the jar file of your command line implementation? My e-mail is nathan.beyne@outlook.com

**Comment by nbeyne1998** [comment]
Has anyone maybe tried implementing this into a asp.net mvc web application, and if one of you did, could you give me some tips to do the same?
